src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -5286,14 +5286,17 @@ internal void RemoveTrailingInQuoteSpaces()
             // Check if the last character is a quote.
             if (ch == '\'' || ch == '\"')
             {
-                if (Char.IsWhiteSpace(Value[i - 1]))
+                if (char.IsWhiteSpace(Value[i - 1]))
                 {
                     i--;
-                    while (i >= 1 && Char.IsWhiteSpace(Value[i - 1]))
+                    while (i >= 1 && char.IsWhiteSpace(Value[i - 1]))
                     {
                         i--;
                     }
-                    Value = Value.Remove(i, Value.Length - 1 - i);
+                    Span<char> result = new char[i + 1];
+                    result[i] = ch;
+                    Value.Slice(0, i).CopyTo(result);
+                    Value = result.AsReadOnlySpan();
```

Comment:
Do we have any tests that hit this and the below removal paths?  It'd be good to make sure they're covered.

Reply:
I will check and add if needed.

Reply:
Added some tests for this particular codepath - https://github.com/dotnet/corefx/pull/27678

Filed an issue for the rest: https://github.com/dotnet/corefx/issues/27679